### PR TITLE
azure active directory authentication support 

### PR DESCRIPTION
**Pull request by alexturkin**
### Problem

in order to be able get data from Azure Insights REST we need to obtain authorization token from Azure Active Directory
### Solution

implement functionality to work with Azure ID via two ways:
- user in AD:  username & password 
- AD application: client id and client secret

### CODE REVIEW

File: src/erlazure_ad_auth.erl [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    ],
+    request(Authority, ADId, Params).
+
+
+
+%%==============================================================================
+%% Internal functions
+%%==============================================================================
+
+request(Authority, ADId, Params) ->
```
Comment (alexturkin) [inline src/erlazure_ad_auth.erl:63]: as @kkuzmin pointed out, we can separate this into some other module in order to be able implement retry/throttling logic in the same place

**Comment by alexturkin** [comment]
@kkuzmin how do you think - does it make sense at all, add oauth-related authorization code? 

Since oauth is standard, some other library could be used for example.
However, this approach has advantage that once you got erlazure, you got everything you need for working with Azure. Also, presense of this module does not force you to use only it in order to obtain auth token.

I am in two minds about it. Thoughts?

**Comment by alexturkin** [comment]
@lilrooness please review this one

**Comment by lilrooness** [comment]
+1

**Pull request merged**